File: store/src/test/java/com/nytimes/android/external/store2/KeyParserTest.java
package com.nytimes.android.external.store2;

import com.nytimes.android.external.store2.base.impl.Store;
import com.nytimes.android.external.store2.base.impl.StoreBuilder;

import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.mockito.runners.MockitoJUnitRunner;

import io.reactivex.Single;
import io.reactivex.observers.TestObserver;


@RunWith(MockitoJUnitRunner.class)
public class KeyParserTest {

    public static final String NETWORK = "Network";
    public static final int KEY = 5;
    private Store<String, Integer> store;

    @Before
    public void setUp() throws Exception {
        store = StoreBuilder.<Integer, String, String>parsedWithKey()
                .parser((integer, s) -> s + integer)
                .fetcher(integer -> Single.just(NETWORK))
                .open();

    }

    @Test
    @SuppressWarnings("PMD.SignatureDeclareThrowsException")
    public void testStoreWithKeyParserFuncNoPersister() throws Exception {
        TestObserver<String> testObservable = store.get(KEY).test().await();
        testObservable.assertNoErrors()
                .assertValues(NETWORK + KEY)
                .awaitTerminalEvent();
    }
}

File: store/src/test/java/com/nytimes/android/external/store3/KeyParserTest.kt
package com.nytimes.android.external.store3

import com.nytimes.android.external.store3.base.impl.Store
import com.nytimes.android.external.store3.base.impl.StoreBuilder
import com.nytimes.android.external.store3.util.KeyParser
import kotlinx.coroutines.runBlocking
import org.assertj.core.api.Assertions.assertThat
import org.junit.Before
import org.junit.Test
import org.junit.runner.RunWith
import org.mockito.runners.MockitoJUnitRunner


@RunWith(MockitoJUnitRunner::class)
class KeyParserTest {
    lateinit var store: Store<String, Int>

    @Before
    @Throws(Exception::class)
    fun setUp() {
        store = StoreBuilder.parsedWithKey<Int, String, String>()
            .parser(object : KeyParser<Int, String, String> {
                override suspend fun apply(key: Int, raw: String): String {
                    return raw + key
                }
            })
            .fetcher { NETWORK }
            .open()
    }

    @Test
    @Throws(Exception::class)
    fun testStoreWithKeyParserFuncNoPersister() = runBlocking<Unit> {
        assertThat(store.get(KEY)).isEqualTo(NETWORK + KEY)
    }

    companion object {

        private const val NETWORK = "Network"
        const val KEY = 5
    }
}
